File: test/fixtures/ruby/put_basic_auth_json_data.rb
require 'net/http'

uri = URI('http://localhost:28139/test/_security')
req = Net::HTTP::Put.new(uri)
req.basic_auth 'admin', '123'
req.content_type = 'application/x-www-form-urlencoded'

req.body = '{"admins":{"names":[], "roles":[]}, "readers":{"names":["joe"],"roles":[]}}'

req_options = {
  use_ssl: uri.scheme == 'https'
}
res = Net::HTTP.start(uri.hostname, uri.port, req_options) do |http|
  http.request(req)
end

File: test/fixtures/php/put_basic_auth_json_data.php
<?php
$ch = curl_init();
curl_setopt($ch, CURLOPT_URL, 'http://localhost:28139/test/_security');
curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
curl_setopt($ch, CURLOPT_CUSTOMREQUEST, 'PUT');
curl_setopt($ch, CURLOPT_HTTPHEADER, [
    'Content-Type: application/x-www-form-urlencoded',
]);
curl_setopt($ch, CURLOPT_HTTPAUTH, CURLAUTH_BASIC);
curl_setopt($ch, CURLOPT_USERPWD, 'admin:123');
curl_setopt($ch, CURLOPT_POSTFIELDS, '{"admins":{"names":[], "roles":[]}, "readers":{"names":["joe"],"roles":[]}}');

$response = curl_exec($ch);

curl_close($ch);
